dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -604,6 +627,37 @@ private ExtensionArchive getProbeExtensionArchive() {
         return this.probeExtensionArchive;
     }
 
+    private ExtensionArchive newSPIExtensionArchive(ServiceReference<WebSphereCDIExtensionMetaData> sr,
+                                                    WebSphereCDIExtensionMetaData webSphereCDIExtensionMetaData, WebSphereCDIDeployment applicationContext) throws CDIException {
+        Bundle bundle = sr.getBundle();
+
+        Set<Class<? extends Extension>> extensionClasses = webSphereCDIExtensionMetaData.getExtensions();
+        Set<Extension> extensions = new HashSet<Extension>();
+        Set<String> extensionClassNames = extensionClasses.stream().map(clazz -> clazz.getCanonicalName()).collect(Collectors.toSet());
+
+        for (Class<? extends Extension> clazz : extensionClasses) {
+            try {
+                extensions.add(clazz.getDeclaredConstructor().newInstance());
+            } catch (InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e) {
+                // Just emit an FFDC. This will proceed without using the extension.
```

Comment:
Are you sure? Throwing a CDIException would be more consistent.

Reply:
I had some memory that we were trying to move away from CDIExceptions. I must have been mistaken